generated/java/google-cloud-pubsub-v1/src/main/java/com/google/cloud/pubsub/spi/v1/SubscriberClient.java

```diff
@@ -1030,6 +1109,339 @@ public final void modifyPushConfig(ModifyPushConfigRequest request) {
+   *     .setProject(formattedProject)
+   *     .build();
+   *   while (true) {
+   *     ListSnapshotsResponse response = subscriberClient.listSnapshotsCallable().call(request);
+   *     for (Snapshot element : response.getSnapshotsList()) {
+   *       // doThingsWith(element);
+   *     }
+   *     String nextPageToken = response.getNextPageToken();
+   *     if (!Strings.isNullOrEmpty(nextPageToken)) {
+   *       request = request.toBuilder().setPageToken(nextPageToken).build();
+   *     } else {
+   *       break;
+   *     }
+   *   }
+   * }
+   * </code></pre>
+   */
+  public final UnaryCallable<ListSnapshotsRequest, ListSnapshotsResponse> listSnapshotsCallable() {
+    return listSnapshotsCallable;
+  }
+
+  // AUTO-GENERATED DOCUMENTATION AND METHOD
+  /**
+   * Creates a snapshot from the requested subscription. If the snapshot already exists, returns
+   * `ALREADY_EXISTS`. If the requested subscription doesn't exist, returns `NOT_FOUND`.
+   *
+   * <p>If the name is not provided in the request, the server will assign a random name for this
+   * snapshot on the same project as the subscription, conforming to the [resource name
+   * format](https://cloud.google.com/pubsub/docs/overview#names). The generated name is populated
+   * in the returned Snapshot object. Note that for REST API requests, you must specify a name in
+   * the request.
+   *
+   * <p>Sample code:
+   *
+   * <pre><code>
+   * try (SubscriberClient subscriberClient = SubscriberClient.create()) {
+   *   String formattedName = SnapshotName.create("[PROJECT]", "[SNAPSHOT]").toString();
+   *   String formattedSubscription = SubscriptionName.create("[PROJECT]", "[SUBSCRIPTION]").toString();
+   *   Snapshot response = subscriberClient.createSnapshot(formattedName, formattedSubscription);
+   * }
+   * </code></pre>
+   *
+   * @param name Optional user-provided name for this snapshot. If the name is not provided in the
+   *     request, the server will assign a random name for this snapshot on the same project as the
+   *     subscription. Note that for REST API requests, you must specify a name. Format is
+   *     `projects/{project}/snapshots/{snap}`.
+   * @param subscription The subscription whose backlog the snapshot retains. Specifically, the
+   *     created snapshot is guaranteed to retain: (a) The existing backlog on the subscription.
+   *     More precisely, this is defined as the messages in the subscription's backlog that are
+   *     unacknowledged upon the successful completion of the `CreateSnapshot` request; as well as:
+   *     (b) Any messages published to the subscription's topic following the successful completion
+   *     of the CreateSnapshot request. Format is `projects/{project}/subscriptions/{sub}`.
+   * @throws com.google.api.gax.grpc.ApiException if the remote call fails
+   */
+  public final Snapshot createSnapshot(String name, String subscription) {
```

Comment:
We need to fix the configuration for pubsub so that we take `SubscriptionName` here.

For expediency, maybe we should skip the pubsub update in this round of updates, because the header updates are time sensitive.